### PR TITLE
Add steps to create and configure a versioned S3 bucket

### PR DESCRIPTION
**Pull request by gursewak1997**
Add tutorial with detailed steps for creating and configuring a versioned S3 bucket, including lifecycle rules for managing versions and delete markers.

### CODE REVIEW

File: docs/s3-bucket-versioning.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# **Steps to Create and Configure a Versioned S3 Bucket**
+
```
Comment (ravanelli) [inline docs/s3-bucket-versioning.md:2]: Would be good to have a summary about what this doc is about here:
Something like:

This file provides a step-by-step guide for creating and configuring an Amazon S3 bucket with versioning and lifecycle management to optimize storage usage and costs.

File: docs/s3-bucket-versioning.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+## **Create a New S3 Bucket**
+Use the following command to create a bucket. Make sure to specify the correct region.
+
+```bash
+aws s3api create-bucket --bucket $BUCKET_NAME --region your-region
+```
+
+---
+
+## **Enable Versioning on the Bucket**
```
Comment (ravanelli) [inline docs/s3-bucket-versioning.md:17]: How can I see/track the changes/versions?
When should I change the version?
Would be good to add a section about it too.

File: docs/s3-bucket-versioning.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
+# **Steps to Create and Configure a Versioned S3 Bucket**
+
+## **Set a Variable for the Bucket Name**
+```bash
+BUCKET_NAME=temp-bucket-for-versioning
```
Comment (ravanelli) [inline docs/s3-bucket-versioning.md:5]: Is that the standard name we use for our bucket?
What is the diff for a normal bucket and this one?

Comment (gursewak1997) [reply docs/s3-bucket-versioning.md:5]: No there's no standard names for the buckets afaik. This is just an example for a name. I'll update the doc specifying the user to pick a name according to the needs.

File: docs/s3-bucket-versioning.md (hunk 2 above)
Comment (gursewak1997) [reply docs/s3-bucket-versioning.md:17]: To track changes/versions, I have added a brief section down in **Verify Configuration**
Version changes happen automatically with each update or delete marker, so manual version changes aren't needed.

**Comment by dustymabe** [comment]
Nice work! I like the explanations.

Maybe here let's also update https://github.com/coreos/fedora-coreos-pipeline/blob/main/HACKING.md#optional-allocating-s3-storage to mention "if you want to version and lifecycle the bucket see this doc for how to do so". and also mention that for FCOS fcos-builds we'll use X days as the lifecycle.

I'm thinking X of 14 days would be appropriate? 2 weeks is easy to remember and should be plenty of time I would think.

**Comment by gursewak1997** [comment]
> I'm thinking X of 14 days would be appropriate? 2 weeks is easy to remember and should be plenty of time I would think.

2 weeks should be ample time to notice any discrepancy in `builds.json` for example. Updated.

File: HACKING.md [hunk 4]
```diff
@@ -242,6 +242,8 @@ If you want to store builds persistently, now is a good time to allocate
 S3 storage.  See the [upstream coreos-assembler docs](https://github.com/coreos/coreos-assembler/blob/main/README-design.md)
 around build architecture.
 
+If you need to set up versioning and lifecycle management for the bucket, refer to [s3 bucket versioning doc](https://github.com/coreos/fedora-coreos-pipeline/blob/main/docs/s3-bucket-versioning.md) for detailed steps. For FCOS builds, we’ll apply a lifecycle policy of 14 days to manage old versions efficiently.
```
Comment (dustymabe) [inline HACKING.md:245]: minor improvement:

```suggestion
If you need to set up versioning and lifecycle management for the bucket, refer to [s3 bucket versioning doc](https://github.com/coreos/fedora-coreos-pipeline/blob/main/docs/s3-bucket-versioning.md) for detailed steps. For FCOS production builds using the `fcos-builds` bucket we apply a lifecycle policy of 14 days to manage old versions efficiently.
```

File: HACKING.md [hunk 5]
```diff
@@ -242,6 +242,8 @@ If you want to store builds persistently, now is a good time to allocate
...
+
```
Comment (dustymabe) [inline HACKING.md:246]: should we update the `aws s3 mb` example below to use the new API? `aws s3api` ?

Comment (gursewak1997) [reply HACKING.md:246]: Obviously, the existing command should work but I am happy to update to new API in another commit.

Comment (dustymabe) [reply HACKING.md:246]: nvm - don't worry about this

Comment (gursewak1997) [reply HACKING.md:246]: Hah, added the changes. Lmk they seem worth the change :eyes:

**Review by ravanelli — APPROVED** [review]
LGTM!

**Review by dustymabe — APPROVED** [review]

**Pull request merged**